bugsnag-android-core/src/main/java/com/bugsnag/android/internal/Journal.kt

```diff
@@ -110,9 +141,28 @@ class Journal {
          * @return The journal
          */
         fun deserialize(bytes: ByteArray): Journal {
-            val journal = Journal()
             var currentStart = 0
-            for (i in bytes.indices) {
+            val length = bytes.size
+
+            // Decode the metadata entry
+            for (i in 0 until length) {
+                if (bytes[i] == 0.toByte()) {
+                    currentStart = i + 1
+                    break
+                }
+            }
+            require(currentStart > 0) { "Serialized journal doesn't contain a metadata entry" }
```

Comment:
nit: these messages should be updated to reword 'metadata' to 'version information' or whatever naming convention is settled upon